near/omni-bridge/src/lib.rs

```diff
@@ -1977,18 +2027,230 @@ impl Contract {
             )
     }
 
+    fn utxo_fin_transfer(
+        &mut self,
+        token_id: AccountId,
+        amount: U128,
+        signer_id: AccountId,
+        sender_id: &AccountId,
+        utxo_fin_transfer_msg: UtxoFinTransferMsg,
+    ) -> PromiseOrPromiseIndexOrValue<U128> {
+        let origin_chain = self
+            .get_utxo_chain_by_token(&token_id)
+            .sdk_expect("ERR_UTXO_CONFIG_MISSING");
+        let config = self
+            .utxo_chain_connectors
+            .get(&origin_chain)
+            .sdk_expect("ERR_UTXO_CONFIG_MISSING");
+        require!(
+            sender_id == &config.connector,
+            "ERR_SENDER_IS_NOT_CONNECTOR"
```

Comment:
So this will only work for safe_deposit, but not for a regular deposit? Are we okay with that?